engines/tot/resources.cpp

```diff
@@ -0,0 +1,395 @@
+	stream->seek(itemPos * kItemRegSize);
+	clearCurrentInventoryObject();
+	thisRegObj.code = stream->readUint16LE();
+	thisRegObj.height = stream->readByte();
+
+	thisRegObj.name = stream->readPascalString();
+
+	stream->skip(kObjectNameLength - thisRegObj.name.size());
+
+	thisRegObj.lookAtTextRef = stream->readUint16LE();
+	thisRegObj.beforeUseTextRef = stream->readUint16LE();
+	thisRegObj.afterUseTextRef = stream->readUint16LE();
+	thisRegObj.pickTextRef = stream->readUint16LE();
+	thisRegObj.useTextRef = stream->readUint16LE();
+	thisRegObj.speaking = stream->readByte();
+	thisRegObj.openable = stream->readByte();
+	thisRegObj.closeable = stream->readByte();
+
+	stream->read(thisRegObj.used, 8);
+
+	thisRegObj.pickupable = stream->readByte();
+	thisRegObj.useWith = stream->readUint16LE();
+	thisRegObj.replaceWith = stream->readUint16LE();
+	thisRegObj.depth = stream->readByte();
+	thisRegObj.bitmapPointer = stream->readUint32LE();
+	thisRegObj.bitmapSize = stream->readUint16LE();
+	thisRegObj.rotatingObjectAnimation = stream->readUint32LE();
+	thisRegObj.rotatingObjectPalette = stream->readUint16LE();
+	thisRegObj.dropOverlayX = stream->readUint16LE();
+	thisRegObj.dropOverlayY = stream->readUint16LE();
+	thisRegObj.dropOverlay = stream->readUint32LE();
+	thisRegObj.dropOverlaySize = stream->readUint16LE();
+	thisRegObj.objectIconBitmap = stream->readUint16LE();
+	thisRegObj.xgrid1 = stream->readByte();
+	thisRegObj.ygrid1 = stream->readByte();
+	thisRegObj.xgrid2 = stream->readByte();
+	thisRegObj.ygrid2 = stream->readByte();
+	stream->read(thisRegObj.walkAreasPatch, 100);
+	stream->read(thisRegObj.mouseGridPatch, 100);
+}
+
+void TotEngine::readObject(uint itemPosition) {
+	readObject(_sceneObjectsData, itemPosition, _curObject);
+}
+
+void TotEngine::updateObject(uint itemPosition) {
+	_curObject.used[0] = 9;
+	_sceneObjectsData->seek(itemPosition);
+	saveObject(_curObject, _sceneObjectsData);
+}
+
+/**
+ * Object files contain a single register per object, with a set of 8 flags, to mark them as used in each save.
+ */
+void TotEngine::initializeObjectFile() {
+	Common::File objFile;
+	if (!objFile.open(Common::Path("OBJETOS.DAT"))) {
+		showError(261);
+	}
+	delete (_sceneObjectsData);
+	byte *objectData = (byte *)malloc(objFile.size());
+	objFile.read(objectData, objFile.size());
+	_sceneObjectsData = new Common::MemorySeekableReadWriteStream(objectData, objFile.size(), DisposeAfterUse::NO);
+	objFile.close();
+}
+
+void TotEngine::saveObjectsData(ScreenObject object, Common::SeekableWriteStream *objectDataStream) {
+	objectDataStream->writeUint16LE(object.code);
+	objectDataStream->writeByte(object.height);
+
+	objectDataStream->writeByte(object.name.size());
+	objectDataStream->writeString(object.name);
+	int paddingSize = kObjectNameLength - object.name.size();
+	if (paddingSize > 0) {
+		char *padding = (char *)malloc(paddingSize);
```

Comment:
```suggestion
		char *padding = (char *)calloc(paddingSize, 1);
```
and then remove the loop below

Reply:
nice!